notify/grafana_alertmanager.go

```diff
@@ -892,6 +901,9 @@ func (am *GrafanaAlertmanager) createReceiverStage(name string, integrations []*
 			Idx:         uint32(integrations[i].Index()),
 		}
 		var s notify.MultiStage
+		if stage := stages.NewSyncFlushStage(notificationLog, recv, syncAct, syncMargin); stage != nil {
```

Comment:
I think this stage should run before Fanout stage, perhaps right after `meshStage`.